pandas/core/indexes/base.py

```diff
@@ -573,7 +573,13 @@ def __array_wrap__(self, result, context=None):
 
         attrs = self._get_attributes_dict()
         attrs = self._maybe_update_attributes(attrs)
-        return Index(result, **attrs)
+        from pandas.core.dtypes.generic import ABCDatetimeIndex
+        if issubclass(self, ABCDatetimeIndex) and 'tz' in attrs:
+            from pandas.core.indexes.datetimes import _new_DatetimeIndex
+            attrs['data'] = result
```

Comment:
instead of making the change here I would like to do this in ``DatetimeIndex`` itself 

e.g.

```
    def __array_wrap__(self, result, context=None):
        """
        Gets called after a ufunc
        """
        if is_bool_dtype(result):
            return result
        attrs = self._get_attributes_dict()
        attrs = self._maybe_update_attributes(attrs)
        return _new_DatetimeIndex(type(self), attrs)
```

Reply:
its a bit more duplicated code, but more clear.